File: 02. Algorithms/09. Dynamic Programming/005. House Robber/code.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int sz = nums.size();
        vector<int> dp(sz);
        for (int i = 0; i < sz; i++) {
            if (i == 0) {
                dp[i] = nums[0];
            } else if (i == 1) {
                dp[i] = max(nums[1], nums[0]);
            } else {
                dp[i] = max(dp[i-1], dp[i-2] + nums[i]);
            }
        }
        return dp[sz-1];
    }
};

File: 02. Algorithms/09. Dynamic Programming/005. House Robber/code.js
var rob = function(nums) {
    
    let rob2 = 0;
    let rob1 = 0;
    
    for(let i = 0; i < nums.length; i++) {
        let temp = Math.max(nums[i] + rob1, rob2);
        rob1 = rob2;
        rob2 = temp; 
    }
    
    return rob2;
};
